internal/rateservice/customers/app/consumer.go

```diff
@@ -0,0 +1,86 @@
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/nats-io/nats.go/jetstream"
+	"go_service/internal/rateservice/customers/presentation"
+	"go_service/internal/rateservice/customers/services/createcustomer"
+	"go_service/internal/rateservice/customers/services/deletecustomer"
+	"go_service/internal/rateservice/infrastructure/broker"
+	"log"
+)
+
+type Message struct {
+	Title         string                 `json:"title"`
+	Type          string                 `json:"type"`
+	TransactionID *string                `json:"transaction_id"`
+	From          string                 `json:"from"`
+	Data          map[string]interface{} `json:"data"`
+}
+
+type Consumer struct {
+	ctx       context.Context
+	js        jetstream.JetStream
+	container presentation.InteractorFactory
+}
+
+func NewConsumer(ctx context.Context, js jetstream.JetStream, container *IoC) Consumer {
+	return Consumer{ctx: ctx, js: js, container: container}
+}
+
+func (c Consumer) Run() jetstream.ConsumeContext {
+	_, _ = broker.NewStream(c.ctx, c.js, "events")
```

Comment:
from my perspective, better to put context to method, not to struct.
https://go.dev/blog/context-and-structs#prefer-contexts-passed-as-arguments